.github/workflows/ecosystem-ci-trigger.yml

```
@@ -22,57 +158,46 @@ jobs:
                 repo: context.repo.repo,
                 username: user,
               });
-              hasTriagePermission = data.user.permissions.triage
+              hasTriagePermission = ['triage', 'write', 'admin'].some(p => data.user.permissions[p]);
             } catch (e) {
               console.warn(e)
             }
 
-            if (hasTriagePermission) {
-              console.log('Allowed')
+            return { allowed: hasTriagePermission }
+
+      - name: React Based on Permissions
+        uses: actions/github-script@v7
+        id: react-permissions
+        with:
+          script: |
+            const allowed = ${{ steps.check-permissions.outputs.allowed }}
+
+            if (allowed) {
```

```suggestion
            if (allowed) {
```
Is there a reason to split these steps? If not, let's merge it.